### PR TITLE
Throw error when using find cursor with sort but no limit

### PR DESCRIPTION
**Pull request by vkarpov15**
**What this PR does**:

Right now stargate-mongoose doesn't offer any feedback if you use sort without limit, it just returns max 20 documents. While that limit is reasonable, I think we should consider throwing an error if using sort without limit. That will provide users more immediate feedback on limitations, so they aren't wondering why they're only getting 20 docs.

Related to #83 

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: tests/collections/collection.test.ts [hunk 1]
```diff
@@ -2173,7 +2174,7 @@ describe(`StargateMongoose - ${testClientName} Connection - collections.collecti
         { sort: { username: 1 } }
       );
 
-      const docs = await collection.find({}, { sort: { username: 1 } }).toArray();
+      const docs = await collection.find({}, { sort: { username: 1 }, limit: 20 }).toArray();
```
Comment (kathirsvn) [inline tests/collections/collection.test.ts:2177]: Can we add a couple of test cases for negative and zero limits as well and have the behavior of JSON API tested?

Comment (vkarpov15) [reply tests/collections/collection.test.ts:2177]: @kathirsvn done in 873d6da . Turns out we treat limit 0 as no limit, which is consistent with how MongoDB and Mongoose handle it.

Negative limits throw an error in stargate-mongoose, which is slightly different than MongoDB. [MongoDB does support negative limits](https://www.mongodb.com/docs/manual/reference/method/cursor.limit/#negative-values); for practical purposes, `limit: -2` is equivalent to `limit: 2`, there's just slightly different behavior in the case where `limit > batchSize`. However, negative limits in MongoDB are rarely used and I don't think we should prioritize supporting negative limits.

**Review by kathirsvn — APPROVED** [review]
